### PR TITLE
feat: improve watcher performance, add forceRerunTriggers option

### PR DESCRIPTION
**Pull request by sheremet-va**
Few things to note here: 

- `watchIgnore` is renamed to `watchExclude`
- `watchExclude` now accepts only glob patterns

> Now, instead of ignoring a file for rerun, we don't even watch for it. This should improve errors in docker about too much watches, and generally improve performance, because no more excessive calls.

- added `forceRerunTriggers` option 
- `forceRerunTriggers` accepts glob pattern for files, that if changed, will cause a full suite rerun
- `forceRerunTriggers` respects `watchExclude`

> It is useful, when running CLI tests, because Vite cannot build a module graph for `exec` calls

### CODE REVIEW

**Comment by antfu** [comment]
What's `dumb` stands for?

**Comment by sheremet-va** [comment]
> What's `dumb` stands for?

"not smart" 👀 

I've recently worked on a project involving dumb devices, so I thought why not 😄  Also, the name was proposed as a joke in discord :P

**Comment by jgoux** [comment]
> > What's `dumb` stands for?
> 
> "not smart" 👀
> 
> I've recently worked on a project involving dumb devices, so I thought why not 😄 Also, the name was proposed as a joke in discord :P

Sorry for this terrible joke. 😂

If the default watch mode is "smart", maybe instead we could have a `--watchMode=smart|dumb`, and if `watchMode` is set to `dumb`, then we could have `watchInclude` taken into account?

File: packages/vitest/src/node/core.ts [hunk 1]
```diff
@@ -122,8 +122,8 @@ export class Vitest {
         this.console.error(c.dim('filter:  ') + c.yellow(filters.join(comma)))
       if (this.config.include)
         this.console.error(c.dim('include: ') + c.yellow(this.config.include.join(comma)))
-      if (this.config.watchIgnore)
-        this.console.error(c.dim('ignore:  ') + c.yellow(this.config.watchIgnore.join(comma)))
+      if (this.config.watchExclude)
+        this.console.error(c.dim('ignore:  ') + c.yellow(this.config.watchExclude.join(comma)))
```
Comment (antfu) [inline packages/vitest/src/node/core.ts:126]:
```suggestion
        this.console.error(c.dim('exclude:  ') + c.yellow(this.config.watchExclude.join(comma)))
```

File: packages/vitest/src/types/config.ts [hunk 2]
```diff
@@ -198,11 +198,18 @@ export interface InlineConfig {
...
+  /**
+   * Glob patter of file paths that will trigger the whole suite rerun
    *
-   * @default [/\/node_modules\//, /\/dist\//]
+   * Useful if you are testing calling CLI commands
+   *
+   * @default []
    */
-  watchIgnore?: (string | RegExp)[]
+  dumbWatchInclude?: string[]
```
Comment (antfu) [inline packages/vitest/src/types/config.ts:212]: Emm, maybe:

- `forceRerunTargets`
- `forceRerunTriggers`
- `forceRerunEntries`

Comment (sheremet-va) [reply packages/vitest/src/types/config.ts:212]: I want to be consistent with "exclude/include"

File: packages/vitest/src/node/core.ts (hunk 1 above)
Comment (sheremet-va) [reply packages/vitest/src/node/core.ts:126]: I think `exclude` is a bit misleading. Renamed to `watch exclude`

**Review by antfu — APPROVED** [review]

**Comment by MichaelDeBoey** [comment]
> `watchIgnore` is renamed to `watchExclude`

This wasn't clear watching only the changelog

**Comment by sheremet-va** [comment]
> > `watchIgnore` is renamed to `watchExclude`
> 
> This wasn't clear watching only the changelog

Sorry, thank you for pointing that out. Will do better next time.

**Pull request merged**